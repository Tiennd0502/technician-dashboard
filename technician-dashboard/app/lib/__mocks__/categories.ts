export const CATEGORIES = [
  {
    id: '01',
    label: 'Sofa Cleaning',
    image: 'https://i.ibb.co/QcSmTHW/category01.webp',
  },
  {
    id: '02',
    label: 'Bathroom',
    image: 'https://i.ibb.co/KDQzZ3s/category02.web',
  },
  {
    id: '03',
    label: 'Best Control',
    image: 'https://i.ibb.co/jwsPTTt/category03.webp',
  },
  {
    id: '04',
    label: 'View All',
    image: 'https://i.ibb.co/KxsM943/category04.webp',
  },
];
