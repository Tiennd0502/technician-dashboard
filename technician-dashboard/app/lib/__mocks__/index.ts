export * from './table';
export * from './categories';
export * from './services';
