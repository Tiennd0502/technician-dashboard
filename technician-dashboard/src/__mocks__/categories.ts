export const CATEGORIES = [
  {
    id: '01',
    label: 'Sofa Cleaning',
    image: 'https://i.ibb.co/dKnNFV1/category01.png',
  },
  {
    id: '02',
    label: 'Bathroom',
    image: 'https://i.ibb.co/Qdtfx8N/category02.png',
  },
  {
    id: '03',
    label: 'Best Control',
    image: 'https://i.ibb.co/dDLCwNf/category03.png',
  },
  {
    id: '04',
    label: 'View All',
    image: 'https://i.ibb.co/cDz3H5Q/category04.png',
  },
];
