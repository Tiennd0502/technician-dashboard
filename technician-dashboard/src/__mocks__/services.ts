export const SERVICE = {
  id: '01',
  name: 'Carpenter',
  price: 45,
  image: 'https://i.ibb.co/7WfPDMJ/service01.png',
};

export const SERVICES = [
  {
    id: '01',
    name: 'Carpenter',
    price: 45,
    image: 'https://i.ibb.co/7WfPDMJ/service01.png',
  },
  {
    id: '02',
    name: 'Ac Service & repair',
    price: 25,
    image: 'https://i.ibb.co/G069RKZ/service02.png',
  },
  {
    id: '03',
    name: 'Kitchecn Deep Cleaning',
    price: 30,
    image: 'https://i.ibb.co/nw3YSHS/service03.png',
  },
  {
    id: '04',
    name: 'Electrician',
    price: 50,
    image: 'https://i.ibb.co/Cvnw8GY/service04.png',
  },
];
