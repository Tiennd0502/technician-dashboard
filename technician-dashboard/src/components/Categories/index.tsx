import { memo, useCallback } from 'react';

import { Category } from '@/interfaces';
import { Link, Flex, Text } from '@chakra-ui/react';

interface CategoriesProps {
  list: Category[];
}

const Categories = ({ list }: CategoriesProps) => {
  const getLinkStyles = useCallback(
    (url: string) => ({
      width: '100%',
      height: '70px',
      borderRadius: 'md',
      background: `linear-gradient(0deg, rgba(0, 0, 0, 0.80) 0%, rgba(0, 0, 0, 0.80) 100%), url(${url}), lightgray 50% / cover no-repeat`,
    }),
    [],
  );

  return (
    <Flex gap='5' role='list'>
      {list.map(({ id, image, label }) => (
        <Link key={id} pos='relative' {...getLinkStyles(image)}>
          <Text
            color='text.tertiary'
            variant='textXsm'
            pos='absolute'
            top='50%'
            left='50%'
            transform='translate(-50%, -50%)'
          >
            {label}
          </Text>
        </Link>
      ))}
    </Flex>
  );
};

export default memo(Categories);
