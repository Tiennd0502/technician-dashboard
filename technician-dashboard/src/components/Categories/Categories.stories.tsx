import type { Meta, StoryObj } from '@storybook/react';

import { CATEGORIES } from '@/__mocks__';

import Categories from '.';

const meta: Meta<typeof Categories> = {
  title: 'Components/Categories',
  component: Categories,
  tags: ['autodocs'],
};

export default meta;
type Story = StoryObj<typeof Categories>;

export const Default: Story = {
  args: {
    list: CATEGORIES,
  },
};
