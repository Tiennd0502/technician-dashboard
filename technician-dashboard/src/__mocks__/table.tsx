import { STATUS } from '@/interfaces';
import { StatusLabel } from '@/components';

export const PRODUCTS = [
  {
    id: '1',
    name: 'Washing machine',
    brand: 'Hitachi',
    service: 'Repair',
    status: 1,
  },
  {
    id: '2',
    name: 'Air conditioner',
    brand: 'LG',
    service: 'Installation',
    status: 0,
  },
  {
    id: '3',
    name: 'Smart TV',
    brand: 'OnePlus',
    service: 'Uninstallation',
    status: 1,
  },
  {
    id: '4',
    name: 'Water purifier',
    brand: 'Kent',
    service: 'Repair',
    status: 1,
  },
];

const customViewStatus = (value: string | number | boolean) => (
  <StatusLabel value={value as STATUS} />
);

export const PRODUCT_HEADER_COLUMNS = [
  {
    key: 'checkbox',
    label: 'Checkbox',
    width: {
      md: '30px',
      lg: '50px',
    },
    isCheckbox: true,
  },
  {
    key: 'name',
    label: 'Product name',
    width: {
      md: '100%',
      lg: '28%',
    },
    onSort: () => null,
  },
  {
    key: 'brand',
    label: 'Brand name',
    width: {
      md: '100%',
      lg: '17%',
    },
  },
  {
    key: 'service',
    label: 'Service',
    width: {
      md: '100%',
      lg: '15%',
    },
  },
  {
    key: 'status',
    label: 'Status',
    width: {
      md: '100%',
      lg: '18%',
    },
    customView: customViewStatus,
  },
  {
    key: 'actions',
    label: 'Actions',
    width: {
      md: '100%',
      lg: 'calc(32% - 35px)',
    },
    isAction: true,
  },
];
