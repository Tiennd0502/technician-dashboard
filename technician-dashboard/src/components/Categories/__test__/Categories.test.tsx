import { render } from '@testing-library/react';

import { CATEGORIES } from '@/__mocks__';
import Categories from '..';

describe('Categories test cases', () => {
  const props = {
    list: CATEGORIES,
  };

  test('should render correctly', () => {
    const { container } = render(<Categories {...props} />);
    expect(container).toMatchSnapshot();
  });

  test('Should render enough elements', () => {
    const { getByRole } = render(<Categories {...props} />);
    const categories = getByRole('list');

    expect(categories.childNodes).toHaveLength(CATEGORIES.length);
  });

  test('Should render empty elements', () => {
    const { getByRole } = render(<Categories list={[]} />);
    const categories = getByRole('list');

    expect(categories.childNodes).toHaveLength(0);
  });
});
