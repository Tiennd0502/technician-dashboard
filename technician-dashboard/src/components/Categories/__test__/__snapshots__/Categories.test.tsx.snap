// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories test cases should render correctly 1`] = `
<div>
  <div
    class="css-1j52cd2"
    role="list"
  >
    <a
      class="chakra-link css-mq8v5c"
    >
      <p
        class="chakra-text css-1kmnq3w"
      >
        Sofa Cleaning
      </p>
    </a>
    <a
      class="chakra-link css-71elz2"
    >
      <p
        class="chakra-text css-1kmnq3w"
      >
        Bathroom
      </p>
    </a>
    <a
      class="chakra-link css-1jgpbug"
    >
      <p
        class="chakra-text css-1kmnq3w"
      >
        Best Control
      </p>
    </a>
    <a
      class="chakra-link css-1katu9w"
    >
      <p
        class="chakra-text css-1kmnq3w"
      >
        View All
      </p>
    </a>
  </div>
</div>
`;
